### ISSUE TITLE
Incorrect data when uploading files

### ISSUE THREAD

**Issue by willsalasare** [781flyingdutchman/background_downloader#164]
The loading speed is not the true one, it reads 993MB/s when I only have 10MB, the progress is incorrect, it goes up to .9 in 2 seconds and the file weighs 2GB, after a few more seconds it fails with the future error already completed

When starting the enqueue method

I'm just using the code from the documentation with no more code to add.
![Screenshot from 2023-10-10 00-27-49](https://github.com/781flyingdutchman/background_downloader/assets/35855979/d94fc80d-f54e-4e02-af09-dfd0c3f1412f)

**Comment by 781flyingdutchman** [comment]
Sorry to hear this. What version of the plugin are you using, and on which platform(s) does this happen (Android, Ios, desktop)? Can you check if this also happens on version 7.9.4?

**Comment by 781flyingdutchman** [comment]
Can you also provide more logs, and the code you use to start the upload and monitor it?

**Comment by willsalasare** [comment]
Thanks for answering, I didn't see that I had answered :)

Android 13 x86_64 working
Desktop ubuntu 23.04  bug

background_downloader: ^7.9.4

Flutter 3.13.7 • channel stable
Tools • Dart 3.1.3 • DevTools 2.25.0

Desktop
Monitor
![image](https://github.com/781flyingdutchman/background_downloader/assets/35855979/366e8a3c-f97d-47f0-95d9-6ac2b0600032)

Enqueue
![image](https://github.com/781flyingdutchman/background_downloader/assets/35855979/5c639fda-787b-4a15-8499-15962c94f462)

Android 
![image](https://github.com/781flyingdutchman/background_downloader/assets/35855979/0d961789-b4e7-442b-923e-00a4b7a8f177)

**Comment by 781flyingdutchman** [comment]
Thanks, I still can't replicate this issue though, sorry. I understand this is for uploads on Desktop - can you confirm this is still an issue?

**Comment by willsalasare** [comment]
Yes, there is still the same problem, I have tried several files and it continues, only on desktop, I tried on Android and they work correctly

**Comment by willsalasare** [comment]
Just try uploading files, but I can try later and let you know

**Comment by 781flyingdutchman** [comment]
Hi, this was a really interesting one, but I think I have it fixed. Can you try it on the `issue_164` branch of the repo?

Change your pubspec.yaml as follows:
```
background_downloader:
    git:
      url: https://github.com/781flyingdutchman/background_downloader.git
      ref: issue_164
```

**Comment by willsalasare** [comment]
The application does not run, it gives me this error

Launching lib/main.dart on Linux in debug mode...
ERROR: ../../../.pub-cache/git/background_downloader-b0587cc94cb1737e271b80f7182fe4843addc951/lib/src/desktop/upload_isolate.dart:75:47: Error: The argument type 'EventSink<List<int>>' can't be assigned to the parameter type 'StreamSink<List<int>>'.
ERROR:  - 'EventSink' is from 'dart:async'.
ERROR:  - 'List' is from 'dart:core'.
ERROR:  - 'StreamSink' is from 'dart:async'.
ERROR:         await transferBytes(inStream, request.sink, fileSize, task, sendPort);
ERROR:                                               ^
ERROR: ../../../.pub-cache/git/background_downloader-b0587cc94cb1737e271b80f7182fe4843addc951/lib/src/desktop/upload_isolate.dart:180:29: Error: The argument type 'EventSink<List<int>>' can't be assigned to the parameter type 'StreamSink<List<int>>'.
ERROR:  - 'EventSink' is from 'dart:async'.
ERROR:  - 'List' is from 'dart:core'.
ERROR:  - 'StreamSink' is from 'dart:async'.
ERROR:           inStream, request.sink, contentLength, task, sendPort);
ERROR:                             ^
ERROR: Target kernel_snapshot failed: Exception
Exception: Build process failed

Exited (1).

**Comment by 781flyingdutchman** [comment]
That's strange. The change that allows this is in the `dart:async` package, and was introduced recently (in version [1.1.0](https://github.com/dart-lang/http/blob/master/pkgs/http/CHANGELOG.md#110), with the commit made only on June 20). However, I am using the same Dart 3.1.3 and Flutter 3.13.7 as you say you are (per above) so I don't understand why this can be different (as far as I know `dart:async` is bundled with each Dart version). 

If you go to line 76 of `upload_isolate.dart` and hover over the `sink` bit in `request.sink`, the type of that should be `StreamSink` and not `EventSink`. If you get `EventSink` (as the error suggests) then somehow you are using an older version of `dart:async`.

Can you confirm you're using Dart 3.1.3 and somehow make sure you are using the right dart:async library?

**Comment by 781flyingdutchman** [comment]
OK, I think I've found it. The http package used is 'external' and listed in  the pubspec.yaml as `http: '>=0.13.0 <2.0.0'`. This recent, necessary change requires you to use version `1.1.0` so I need to change the pubspec.yaml to say `http: '^1.1.0` which is OK (the wider range was mostly to accommodate Dart 2.x and for the version 7 of teh downloader you need Dart 3 anyway).

So, if you can upgrade using `flutter pub upgrade` you should go to version 1.1.0 and it should run. Let me know if this works.

**Comment by willsalasare** [comment]
Now it works very well, thank you very much.

**Comment by 781flyingdutchman** [comment]
Fixed in V7.11.1
